v2/internal/resolver/config_map_resolver.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package resolver
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+// ConfigMapResolver is a configmap resolver
+type ConfigMapResolver interface {
+	ResolveConfigMapReference(ctx context.Context, ref genruntime.NamespacedConfigMapReference) (string, error)
+	ResolveConfigMapReferences(ctx context.Context, refs set.Set[genruntime.NamespacedConfigMapReference]) (genruntime.Resolved[genruntime.ConfigMapReference], error)
+}
+
+// kubeConfigMapResolver resolves Kubernetes config maps
+type kubeConfigMapResolver struct {
+	client kubeclient.Client
+}
+
+var _ ConfigMapResolver = &kubeConfigMapResolver{}
+
+func NewKubeConfigMapResolver(client kubeclient.Client) ConfigMapResolver {
+	return &kubeConfigMapResolver{
+		client: client,
+	}
+}
+
+// ResolveConfigMapReference resolves the configmap reference and returns the corresponding value, or an error
+// if it could not be found
+func (r *kubeConfigMapResolver) ResolveConfigMapReference(ctx context.Context, ref genruntime.NamespacedConfigMapReference) (string, error) {
+	refNamespacedName := types.NamespacedName{
+		Namespace: ref.Namespace,
+		Name:      ref.Name,
+	}
+
+	configMap := &v1.ConfigMap{}
+	err := r.client.Get(ctx, refNamespacedName, configMap)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err := NewConfigMapNotFoundError(refNamespacedName, err)
+			return "", errors.WithStack(err)
+		}
+
+		return "", errors.Wrapf(err, "couldn't resolve config map reference %s", ref.String())
+	}
+
+	value, ok := configMap.Data[ref.Key] // TODO: Do we need to also check the binaryData field?
+	if !ok {
+		return "", NewSecretNotFoundError(refNamespacedName, errors.Errorf("ConfigMap %q does not contain key %q", refNamespacedName.String(), ref.Key))
```

Comment:
`NewSecretNotFoundError` maybe we need one for configmap? or a generic one for both?

Reply:
Oh I see you got `NewConfigMapNotFoundError` already.

Reply:
Ooh good catch, this should be using the new error type.